crypto/rsa/rsa_ossl.c

```diff
@@ -589,14 +589,26 @@ static int rsa_ossl_public_decrypt(int flen, const unsigned char *from,
     return r;
 }
 
-static int rsa_ossl_mod_exp(BIGNUM *r0, const BIGNUM *I, RSA *rsa, BN_CTX *ctx)
+static int rsa_ossl_mod_exp(BIGNUM *r0, const BIGNUM *Iconst, RSA *rsa, BN_CTX *ctx)
 {
+    /*
+     * If used directly, `Iconst` can be modified by a low level fault attack
+     * injection during operations in this function.It should be cleared in
+     * this case. However, the const flag prevents us to do so. For this
+     * reason, we work with a copy `I` of `Iconst`.
+     * By this way, `I` can be faulted instead of `Iconst` and `I` can be
+     * cleared.
+     * `I` is then a sensitive variable that is cleared at the end of this
+     * function.
+     * */
+    BIGNUM *I;
     BIGNUM *r1, *m1, *vrfy, *r2, *m[RSA_MAX_PRIME_NUM - 2];
     int ret = 0, i, ex_primes = 0, smooth = 0;
     RSA_PRIME_INFO *pinfo;
 
     BN_CTX_start(ctx);
-
+    
+    I = BN_dup(Iconst);
```

Comment:
Other than the BN_CTX_get below, BN_dup may fail because of OOM,
so the result has to be checked to be != NULL